### PR TITLE
Sidebar plugin

### PR DESCRIPTION
**Pull request by felixfontein**
This plugin generates a sidebar include for every language based on a template.

### CODE REVIEW

File: v7/sidebar/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+This plugin renders a sidebar ``output/sidebar-LANG.inc`` defined by a template ``sidebar.tmpl``. The sidebar can contain the most recent posts, all categories, all tags, the archives, all sections, and other taxonomies and arbitrary other content. The packaged ``sidebar.tmpl`` and ``sidebar-helper.tmpl`` shows how this information can be shown.
+
+The generated include file, one per language, must be somehow included in the rest of the blog. This can be done by using JavaScript to dynamically include the file, or by using tools like `File Tree Subs <https://github.com/felixfontein/filetreesubs/>`__.
+
+The number of most recent posts can be changed by defining ``SIDEBAR_MAXIMAL_POST_COUNT = number`` in the Nikola configuration.
```
Comment (Kwpolska) [inline v7/sidebar/README.md]: maximum sounds better

File: v7/sidebar/sidebar.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+                              node.indent_levels, node.indent_change_before,
+                              node.indent_change_after,
+                              len(node.children),
+                              len([post for post in posts_per_tag[node.classification_name] if acceptor(post)]))
+                             for node in flat_hierarchy]
+            else:
+                # Fake hierarchy
+                hierarchy = [(classification_name, classification, taxonomy.extract_hierarchy(classification),
+                              self.site.link(taxonomy_name, classification, lang),
+                              (i, len(result)), 1 if i == 0 else 0, -1 if i == len(result) - 1 else 0, 0,
```
Comment (Kwpolska) [inline v7/sidebar/sidebar.py]: That’s pretty cryptic, don’t you think?

File: v7/sidebar/templates/jinja/sidebar-helper.tmpl [hunk 3]
```diff
@@ -0,0 +1,55 @@
+{# -*- coding: utf-8 -*- #}
+
+{% macro posts(posts) %}
```
Comment (Kwpolska) [inline v7/sidebar/templates/jinja/sidebar-helper.tmpl]: That name might easily conflict with the `posts` variable.

File: v7/sidebar/README.md (hunk 1 above)
Comment (felixfontein) [inline v7/sidebar/README.md]: fixed

File: v7/sidebar/templates/jinja/sidebar-helper.tmpl (hunk 3 above)
Comment (felixfontein) [inline v7/sidebar/templates/jinja/sidebar-helper.tmpl]: I renamed it to `list_posts`.

File: v7/sidebar/sidebar.py (hunk 2 above)
Comment (felixfontein) [inline v7/sidebar/sidebar.py]: I'll document it better

Comment (felixfontein) [inline v7/sidebar/sidebar.py]: I changed the code instead.

**Pull request merged**